Source/WebCore/svg/SVGAnimationElement.cpp

```diff
@@ -147,6 +148,20 @@ bool SVGAnimationElement::isSupportedAttribute(const QualifiedName& attrName)
     return supportedAttributes.get().contains<SVGAttributeHashTranslator>(attrName);
 }
 
+bool SVGAnimationElement::isJavaScriptURLAttribute(const Attribute& attribute) const
+{
+    if (attribute.name() == SVGNames::valuesAttr) {
+        for (auto innerValue : StringView(attribute.value()).split(';')) {
+            if (WTF::protocolIsJavaScript(innerValue.stripWhiteSpace()))
+                return true;
+        }
+        return false;
+    }
+    if (attribute.name() == SVGNames::fromAttr || attribute.name() == SVGNames::toAttr)
+        return WTF::protocolIsJavaScript(stripLeadingAndTrailingHTMLSpaces(attribute.value()));
+    return false;
+}
+
```

Comment:
I am still having trouble reading this function. If it returns true this does not necessarily mean the attribute is a JavaScriptURL since we do not check for the attributeName='href'. 

What is more trouble some is `SVGAnimationElement::isJavaScriptURLAttribute()` may return true while `SVGAnimationElement::isURLAttribute() `may return false for the same attribute. 

I think this function returns whether the attribute **contains** a JavaScriptURL or not. So how about adding a new virtual function called `Element::attributeContainsJavaScriptURL()`?

_// Element.cpp_

```
bool Element::isJavaScriptURLAttribute(const Attribute& attribute) const
{
    return (isURLAttribute(attribute)
        && WTF::protocolIsJavaScript(stripLeadingAndTrailingHTMLSpaces(attribute.value())))
        || attributeContainsJavaScriptURL(attribute);
}
```

_// Element.h_

```
    virtual bool isURLAttribute(const Attribute&) const { return false; }
    virtual bool attributeContainsURL(const Attribute& attribute) const { return isURLAttribute(attribute); }
    virtual bool attributeContainsJavaScriptURL(const Attribute& attribute) const { return false;
```

_// SVGAnimationElement.cpp_

```
bool SVGAnimationElement::attributeContainsJavaScriptURL(const Attribute& attribute) const
{
    if (attribute.name() == SVGNames::valuesAttr) {
        ...
    }
    if (attribute.name() == SVGNames::fromAttr || attribute.name() == SVGNames::toAttr)
        ...
    return false;
}
```

I think this makes sense to me since this function returns exactly what its name is saying.

Reply:
I renamed isJavascriptURL to attributeContainsJavascriptURL